src/main/java/edu/ie3/datamodel/io/naming/FileNamingStrategy.java

```diff
@@ -0,0 +1,248 @@
+      Optional<String> getDirectoryPath(T timeSeries) {
+    Optional<String> maybeDirectoryName = directoryHierarchy.getSubDirectory(timeSeries.getClass());
+    String directoryPath;
+    if (!maybeDirectoryName.isPresent()) {
+      logger.debug("Cannot determine directory name for time series '{}'.", timeSeries);
+      return Optional.empty();
+    } else {
+      /* Make sure, the directory path does not start or end with file separator and in between the separator is harmonized */
+      directoryPath =
+          IoUtil.harmonizeFileSeparator(
+              maybeDirectoryName
+                  .get()
+                  .replaceFirst("^" + IoUtil.FILE_SEPARATOR_REGEX, "")
+                  .replaceAll(IoUtil.FILE_SEPARATOR_REGEX + "$", ""));
+      return Optional.of(directoryPath);
+    }
+  }
+
+  public Pattern getIndividualTimeSeriesPattern() {
+    String subDirectory = directoryHierarchy.getSubDirectory(IndividualTimeSeries.class).orElse("");
+
+    if (subDirectory.isEmpty()) {
+      return entityNamingStrategy.getIndividualTimeSeriesPattern();
+    } else {
+      /* Build the pattern by joining the sub directory with the file name pattern, harmonizing file separators and
+       * finally escaping them */
+      String joined =
+          FilenameUtils.concat(
+              subDirectory, entityNamingStrategy.getIndividualTimeSeriesPattern().pattern());
+      String harmonized = IoUtil.harmonizeFileSeparator(joined);
+      String escaped = harmonized.replace("\\", "\\\\");
+
+      return Pattern.compile(escaped);
+    }
+  }
+
+  public Pattern getLoadProfileTimeSeriesPattern() {
+    String subDirectory = directoryHierarchy.getSubDirectory(LoadProfileInput.class).orElse("");
+
+    if (subDirectory.isEmpty()) {
+      return entityNamingStrategy.getLoadProfileTimeSeriesPattern();
+    } else {
+      /* Build the pattern by joining the sub directory with the file name pattern, harmonizing file separators and
+       * finally escaping them */
+      String joined =
+          FilenameUtils.concat(
+              subDirectory, entityNamingStrategy.getLoadProfileTimeSeriesPattern().pattern());
+      String harmonized = IoUtil.harmonizeFileSeparator(joined);
+      String escaped = harmonized.replace("\\", "\\\\");
+
+      return Pattern.compile(escaped);
+    }
+  }
+
+  /**
+   * Extracts meta information from a file name, of a time series. Here, a file name <u>without</u>
+   * leading path has to be provided
+   *
+   * @param fileName File name
+   * @return The meeting meta information
+   */
+  public FileNameMetaInformation extractTimeSeriesMetaInformation(String fileName) {
+    return entityNamingStrategy.extractTimeSeriesMetaInformation(fileName);
+  }
+
+  /**
+   * Get the entity name for coordinates
+   *
+   * @return the entity name string
+   */
+  public String getIdCoordinateEntityName() {
+    return entityNamingStrategy.getIdCoordinateEntityName();
```

Comment:
I think this somehow is missing a directory handling, isn't it?

Reply:
I don't really understand the use case of this function, there were also no tests for it in `EntityPersistenceNamingStrategy`?

What is that function for and what would the directory be for that one? As far as I can see it's also not covered in `DefaultDirectoryHierarchy`

Reply:
It is used solely here:

https://github.com/ie3-institute/PowerSystemDataModel/blob/0a2c5ef92a1ae52af0bc343f6ff5f953c519c696/src/main/java/edu/ie3/datamodel/io/connectors/CsvFileConnector.java#L330-L340

That is the place, where a file reader of the mapping of coordinate id's towards actual geographical coordinates of weather values is made.

From my point of view, as it does not belong to real "model" we have, we should not integrate it into the `DirectoryHierarchy`. Maybe we could expose the `getBaseDirectory` to the interface `FileHierarchy` and prepend that information to the value of `getIdCoordinateEntityName`.

Reply:
Okay I think I understood the idea.
But I'm not sure how I can make this work, for the `DefaultDirectoryHierarchy` there is a `baseDirectory` string passed in the constructor which I can use, for the `FlatDirectoryHierarchy` I used an empty constructor without a `baseDirectory` until now -> Is this correct like that, or should a `baseDirectory` also be provided for the empty hierarchy?